core/commonTest/src/contract/set/ImmutableSetTest.kt

```diff
@@ -10,13 +10,50 @@ import tests.contract.compare
 import tests.contract.setBehavior
 import tests.isDigit
 import tests.isUpperCase
+import tests.stress.IntWrapper
 import kotlin.test.*
 
 class ImmutableHashSetTest : ImmutableSetTestBase() {
     override fun <T> immutableSetOf(vararg elements: T) = persistentHashSetOf(*elements)
     override fun <T> testBuilderToPersistentSet(builder: PersistentSet.Builder<T>) {
         assertNotSame(builder.build(), builder.toPersistentSet(), "toPersistent shouldn't call build()")
     }
+
+    @Test fun addAllElements() {
+        run {
+            val left = immutableSetOf<Int>() + (1..2000)
+            compareSets(left, left.addAll(immutableSetOf()))
+            compareSets(left, immutableSetOf<Int>().addAll(left))
```

Comment:
Good point

Reply:
Though the second one is not required to be reference equal. That is `empty + pSet` is not required to be `=== pSet`.

Reply:
> That is empty + pSet is not required to be === pSet.

It is not required (or true) with current impl, it will be true with mine afaik, but conceptually I think it should not be required.